src/cgame/cg_weapons.cpp

```diff
@@ -2189,16 +2189,68 @@ void CG_DrawItemSelectText()
 	Rocket_SetPropertyById( "", "opacity", va( "%f", alpha ) );
 }
 
+/*
+===============
+CG_FindNextWeapon
+Find next weapon in inventory.
+===============
+*/
+weapon_t CG_FindNextWeapon( playerState_t *ps )
+{
+	weapon_t nextWeapon = WP_NONE;
+
+	// For each weapon in the known weapon list from the beginning to the end:
+	for ( weapon_t w = WP_NONE; w < WP_NUM_WEAPONS; w = weapon_t( w + 1 ) )
```

Comment:
@slipher I guess we can also start from `WP_NONE + 1`, what do you think about it?

Reply:
That's OK but not too important. Using w = WP_NUM_WEAPONS is much worse, since some things use the weapon number as an index into an array of length WP_NUM_WEAPONS.

Reply:
Yes, in this case that's just an optimization (saving one loop) but not a mistake by itself.